### PR TITLE
Overhaul for modern python+mopidy

### PR DESCRIPTION
**Pull request by djmattyg007**
- Support python3
- Use standard mopidy code quality tools
- Replace travis with github actions
- Rewrite tests to use pytest
- Add initial type checking
- Modify configuration handling slightly
- Other minor tweaks
- Update readme
  - Reformat to make it easier to read outside of github
  - Update links
  - Remove changelog from readme. It's of little value in its current
    form, and should be continued as a standalone file if still deemed
    necessary in future.

### CODE REVIEW

**Comment by djmattyg007** [comment]
A couple of notes:

- Releases to PyPI should happen automatically when a Github Release is created based on a tag. So once this is merged, you'll need to create a tag for `1.0.0`, push it to Github, then create a Github release.
- Before you do that, you'll need to add a secret to this repo with the label `PYPI_TOKEN`. You'll need to generate this token on PyPI, and ideally it should be scoped to just `Mopidy-Webhooks`. This is compulsory for performing releases.
- I've set up the repo with a tool named `invoke`, which is a python task runner. In case you haven't come across it before, you can find some docs here: http://docs.pyinvoke.org/ I've found it to be a great way of scripting common tasks in python repos.

**Comment by djmattyg007** [comment]
Also it looks like you'll need to enable Github Actions on this repo so that the workflows can run at all.

File: README.rst
Comment (djmattyg007) [inline README.rst]: This probably needs updating to point at codecov, but you'd need to sign up there to get the coverage data flowing.

File: mopidy_webhooks/reporters/status.py
Comment (djmattyg007) [inline mopidy_webhooks/reporters/status.py:62]: Hopefully when this PR is merged, this code can either be replaced with a single call to `core.get_status()`, or a second function named something like "full status" could be added to do that instead.

https://github.com/mopidy/mopidy/pull/2007

**Comment by paddycarey** [comment]
> Also it looks like you'll need to enable Github Actions on this repo so that the workflows can run at all.

I checked and they're already enabled, do they need enabled in your forked repo perhaps?

File: setup.cfg
Comment (paddycarey) [inline setup.cfg]: should this still point at the source repo?

**Review by paddycarey — COMMENTED** [review]
Sorry for the delay, I've been on vacation. Just one small comment, but otherwise looks great, thanks!

Let me figure out how to make the tests run and then we can get it merged and released.

File: setup.cfg
Comment (djmattyg007) [reply setup.cfg]: It should 😅

**Comment by djmattyg007** [comment]
I've updated the PR. Hopefully it's good to go now.

I suspect the reason the CI workflow isn't running yet is because of some changes Github made recently to prevent abuse of Github Actions. As I haven't contributed to this repo before, and the workflow didn't exist before this PR, I doubt it will run.

I think it will be easier to merge this PR first, then worry if the workflow doesn't start running after that. If it doesn't work, I'll be more than happy to submit another PR to make it work :)

File: README.rst
Comment (paddycarey) [reply README.rst]: Let's remove codecov for now, and once the action is running post-merge we can add it back (just trying to minimise the number of new things being merge at once).

**Review by paddycarey — APPROVED** [review]
LGTM (once the codecov comment above is addressed, will merge once that's taken care of)

**Comment by djmattyg007** [comment]
I've updated the PR with the following changes:

- Removed the codecov step from the CI workflow
- Removed the codecov badge from the readme
- Removed the old URL target for the CI badge in the readme

**Comment by paddycarey** [comment]
@djmattyg007 merged, thanks! sorry for the delay in responses here, been a busy few weeks 😅

**Comment by paddycarey** [comment]
looks like there were a few warnings in the actions run, but it completed successfully https://github.com/paddycarey/mopidy-webhooks/actions/runs/1214740261

**Comment by djmattyg007** [comment]
I've been seeing similar warnings with Python 3.10 in other repos. Not sure what's going on :shrug: But 3.10-rc2 came out recently, so I'll submit another PR to update the workflow and I guess we'll see if that fixes it.

**Pull request merged**
